### PR TITLE
Remove Inheritance Style Constraint Stings from Schemas

### PR DESCRIPTION
**Pull request by MattToast**
Removes the Inheritance style constraint strings from schemas. Uses `t.Annotated` style pydantic constraints (pydantic is unable to validate when using `= Fields(...)` constraints when there are multiple fileds in the type for which the constraint can apply, and I wanted to try and be consistent across the code base).

## WARNING:
`t.Annotated` was introduced in Python 3.9! This change is incompatible with python 3.8, but seeing as dragon does not support any python versions less than 3.9 this should not be an issue. If needed, we can consider depending on `typing_extensions` to supply `Annotated`.

### CODE REVIEW

**Review by al-rigazzi — CHANGES_REQUESTED** [review]
Looks great! Since we are most likely going to do this anyway, would you mind also removing the Python 3.8 entry from the CI/CD matrix? I prefer seeing green check marks.

**Comment by MattToast** [comment]
Can do! For now I'll just exclude 3.8 from the CI, but I also threw up a [ticket here](https://github.com/CrayLabs/Poseidon/issues/663) to make a more comprehensive change across all of the Smart* repos.

**Review by al-rigazzi — APPROVED** [review]
LGTM, awaiting @ankona's OK

File: .github/workflows/run_tests.yml [hunk 1]
```diff
@@ -57,7 +57,7 @@ jobs:
         os: [macos-12, ubuntu-20.04] # Operating systems
         compiler: [8] # GNU compiler version
         rai: [1.2.7] # Redis AI versions
-        py_v: ['3.8', '3.9', '3.10', '3.11'] # Python versions
+        py_v: ['3.9', '3.10', '3.11'] # Python versions
```
Comment (ankona) [inline .github/workflows/run_tests.yml:60]: If we make this change, should it be [mirrored](https://github.com/CrayLabs/SmartRedis/blob/develop/.github/workflows/run_tests.yml#L67) in SmartRedis?

Comment (ankona) [reply .github/workflows/run_tests.yml:60]: ditto in `release` [action](https://github.com/al-rigazzi/SmartSim/blob/protodrg/.github/workflows/release.yml#L101)

Comment (ankona) [reply .github/workflows/run_tests.yml:60]: ditto in [docs](https://github.com/CrayLabs/SmartSim/blob/develop/doc/installation_instructions/basic.rst?plain=1#L23)

Comment (ankona) [reply .github/workflows/run_tests.yml:60]: Derp. Now i see earlier comments covering this... Carry on, nothing to see here!

**Review by ankona — APPROVED** [review]
LGTM!

**Pull request merged**